### PR TITLE
feat(app): add deck config option to device reset

### PR DESCRIPTION
**Pull request by koji**
# Overview
Add a new reset option, deck configuration to desktop and odd. 
This pr will fix odd device reset screen's UI issue.

For deck configuration reset, the design team decided to include that option into all stored data only. We don't need to add a new checkbox for deck configuration which means no design changes in this PR. In terms of the text of all stored data, Ed updated the text. All information is the following Slack thread.
https://opentrons.slack.com/archives/CSCLVUW3C/p1701889363390079
The text update got approval from Ed. 

[before]
Screenshot 2023-12-06 at 1 51 11 PM

[after]
Screenshot 2023-12-07 at 11 09 34 AM

close RAUT-895

# Test Plan
1. update deck configuration on Desktop App/ODD
2. check the endpoint `/deck_configuration` and `lastModifiedAt` is updated

[Desktop]
go to Robot Settings on Desktop App and check clear all stored data.
then check the endpoint. 

[ODD]
go to Robot Settings on ODD App and check clear all stored data.
then check the endpoint.

# Changelog
- update DeviceReset component and its test
- update DeviceReset Slideout component and its test
- update the text for clear all stored data

# Review requests

# Risk assessment
low

### CODE REVIEW

**Review by ecormany — COMMENTED** [review]
Text changes look good! 👍

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 1]
```diff
@@ -159,16 +165,20 @@ export function DeviceReset({
...
       setResetOptions({
         ...resetOptions,
         authorizedKeys: false,
         onDeviceDisplay: false,
+        deckConfiguration: false,
       })
     }
   }, [resetOptions])
 
+  console.log(resetOptions)
```
Comment (jerader) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: forgotten console.log

**Review by jerader — APPROVED** [review]
lgtm, left a clean up comment

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx (hunk 1 above)
Comment (koji) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: thanks!

**Pull request merged**